### PR TITLE
[Python] CommissionOnNetwork blocked if DUT does not have an open commissoning window

### PR DESCRIPTION
**Pull request by tianfeng-yang**
If DUT does not have an open commissoning window. 
Call Python API `CommissionOnNetwrok` will be blocked.

### CODE REVIEW

File: src/controller/python/ChipDeviceController-ScriptPairingDeviceDiscoveryDelegate.cpp [hunk 1]
```diff
@@ -32,6 +32,10 @@ void ScriptPairingDeviceDiscoveryDelegate::OnDiscoveredDevice(const Dnssd::Disco
     nodeData.resolutionData.ipAddress[0].ToString(buf);
     ChipLogProgress(chipTool, "Discovered Device: %s:%u", buf, port);
 
+    // Cancel discovery timer.
+    if (mActiveDeviceCommissioner->GetSystemLayer() != nullptr)
```
Comment (tcarmelveilleux) [inline src/controller/python/ChipDeviceController-ScriptPairingDeviceDiscoveryDelegate.cpp:36]: Is there not a better place to get the system layer than adding a getter to CHIPDeviceController?

File: src/lib/core/CHIPConfig.h [hunk 2]
```diff
@@ -1345,6 +1345,18 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_CONFIG_SETUP_CODE_PAIRER_DISCOVERY_TIMEOUT_SECS 30
 #endif // CHIP_CONFIG_SETUP_CODE_PAIRER_DISCOVERY_TIMEOUT_SECS
 
+/**
+ *  @def CHIP_CONFIG_ON_NETWORK_PAIRER_DISCOVERY_TIMEOUT_SECS
+ *
+ *  @brief
+ *    This is the default timeout for the discovery of devices by
+ *    the on network pairer.
```
Comment (tcarmelveilleux) [inline src/lib/core/CHIPConfig.h:1353]: This is too generic. This should be settable via the Python API. It should not be a compile-time constant, and should not be here if it only applies to Python controller.

Comment (tianfeng-yang) [reply src/lib/core/CHIPConfig.h:1353]: In fact, this problem also occurs on other platforms, but the chip-tool demo limits the default timeout period of 2 minutes for command execution.
But you are right, it should be more reasonable as a parameter

File: src/controller/python/ChipDeviceController-ScriptPairingDeviceDiscoveryDelegate.cpp (hunk 1 above)
Comment (tianfeng-yang) [reply src/controller/python/ChipDeviceController-ScriptPairingDeviceDiscoveryDelegate.cpp]: Now I am using `chip::DeviceLayer::SystemLayer()`

**Comment by woody-apple** [comment]
@tcarmelveilleux Can you please review?

**Pull request merged**